### PR TITLE
Support generating dictionary intializers.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide  for the updated functionality.
Tagging @dotnet/roslyn-compiler  for the small parser change and the small API addition.

**Comment by cston** [comment]
Compiler changes LGTM.

File: src/EditorFeatures/CSharpTest/UseCollectionInitializer/UseCollectionInitializerTests.cs [hunk 1]
```diff
@@ -50,6 +50,184 @@ void M()
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCollectionInitializer)]
+        public async Task TestIndexAccess1()
```
Comment (jmarolf) [inline src/EditorFeatures/CSharpTest/UseCollectionInitializer/UseCollectionInitializerTests.cs]: >TestIndexAccess1 [](start = 26, length = 16)

Shouldn't we restrict this feature to only work in C#6 or newer?

**Comment by CyrusNajmabadi** [comment]
Ping @jaredpar because there's a tiny public addition here.  This PR adds an extension method "WithoutTrivia(this SyntaxToken)".  This is just a peer to the existing WithoutTrivia we already have that operates on nodes.

**Comment by jaredpar** [comment]
👍 for the API change

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler  Can i get a second pair of eyes on the very tiny change to LanguageParser.cs

**Pull request merged**